tiledb/CMakeLists.txt

```diff
@@ -700,7 +715,13 @@ set(TILEDB_EXPORT_HEADER "${CMAKE_CURRENT_BINARY_DIR}/tiledb_export.h")
 set(TILEDB_EXPORT_HEADER "${TILEDB_EXPORT_HEADER}" PARENT_SCOPE)
 
 # Set related compiler settings
-target_compile_definitions(TILEDB_CORE_OBJECTS PRIVATE -DTILEDB_CORE_OBJECTS_EXPORTS)
+if(WIN32)
+  target_compile_definitions(TILEDB_CORE_OBJECTS PRIVATE -DTILEDB_CORE_OBJECTS_EXPORTS -DCURL_STATICLIB -Dazure_storage_lite_EXPORTS)
```

Comment:
This should at least be conditional on `TILEDB_AZURE`, but ideally would be already handled by the target here: https://github.com/TileDB-Inc/TileDB/blob/7e1c6df4ee932a2b2393fee07cb821bbb2ca2569/tiledb/CMakeLists.txt#L315-L318